src/Buttons/ButtonDropdown.tsx

```diff
@@ -0,0 +1,166 @@
+   */
+  options: Array<DropdownOptionProps>;
+  /** Specifies the alt text for screen readers */
+  alt: string;
+  /** Specifies the ref of the element */
+  forwardRef?: Ref<HTMLButtonElement>;
+}
+
+const StyledButtonDropdown = styled.button`
+  background: ${fromTheme('color', 'background', 'medium')};
+  border: none;
+  cursor: pointer;
+  display: inline-block;
+  font-size: ${fromTheme('font', 'body', 'size')};
+  font-weight: ${fromTheme('font', 'body', 'weight')};
+  padding: ${({ theme }) => (`${theme.ws.xsmall} ${theme.ws.small}`)};
+  &:hover {
+    background: ${fromTheme('color', 'background', 'dark')};
+    border: none;
+    color: ${fromTheme('color', 'text', 'base')};
+  }
+`;
+
+const StyledMenu = styled.div<MenuProps>`
+  background: ${fromTheme('color', 'background', 'light')};
+  border: ${fromTheme('border', 'hairline')};
+  min-height: 3em;
+  min-width: 6em;
+  position: absolute;
+  transform: translate3d(0em, 0.1em, 0em);
+  z-index: 100;
+`;
+
+const StyledMenuList = styled.ul`
+  list-style-type: none;
+`;
+
+const StyledMenuListItem = styled.li`
+  padding: .25em 1em;
+  &:hover {
+    background: ${fromTheme('color', 'background', 'medium')};
+    cursor: pointer;
+  }
+`;
+
+const StyledMenuButton = styled.button`
+  background: transparent;
+  border: none;
+  &:hover {
+    background: ${fromTheme('color', 'background', 'medium')};
+    cursor: pointer;
+  }
+`;
+
+/**
+ * A component that allows users to specify a Font Awesome icon to be displayed
+ * in the button and options within the dropdown. When an option is clicked, the
+ * value that was selected is passed to the onChange function and the dropdown
+ * menu collapses.
+ */
+const ButtonDropdown: FunctionComponent<ButtonDropdownProps> = (props)
+: ReactElement => {
+  const {
+    id,
+    children,
+    onChange,
+    forwardRef,
+    alt,
+    options,
+  } = props;
+
+  const [isMenuVisible, setMenuVisible] = useState(false);
+  const theme = useContext(ThemeContext);
+
+  // When the menu is open and the user clicks on the document, it minimizes the menu
+  const handleDocumentBodyClick = useCallback(() => {
+    setMenuVisible(false);
+  }, [setMenuVisible]);
+
+  // Removes the event listener after the click event occurs
+  useEffect(() => {
+    if (isMenuVisible) {
+      document.addEventListener('click', handleDocumentBodyClick);
+    } else {
+      document.removeEventListener('click', handleDocumentBodyClick);
+    }
+  }, [isMenuVisible, handleDocumentBodyClick]);
+
+  return (
+    <>
+      <StyledButtonDropdown
+        id={id}
+        onClick={() => {
+          setMenuVisible(!isMenuVisible);
+        }}
+        theme={theme}
+        ref={forwardRef}
+        aria-label={alt}
+      >
+        { children }
+      </StyledButtonDropdown>
+      {isMenuVisible && (
+        <>
+          <StyledMenu isMenuVisible={isMenuVisible}>
+            <StyledMenuList>
+              {options.map((option) => (
```

Comment:
Just thinking about this some more, I feel like this is a bit close to how we do our `TabList` than we do the `Dropdown`. And with the tabs, we expose both the `TabList` and the `TabListItem` and let the user compose them however they want. I wonder if something more like that would make more sense here, where we could expose `ButtonDropdownMenu` and `ButtonDropdownMenuItem` components and let the application handle looping through them. My thought there is that:

1) We don't have to be as prescriptive about the `onClick` handler or the shape of the `options`, so in the case of our time picker we could, in course-planner, have something like `{ label: '1:00 PM - 2:00PM', startTime: '13:00:00', endTime: '14:00:00 }` and handle those separately.

2) It would potentially let us put non-clickable items (i.e. headers) in the menu.

I may be overthinking things at this point, but wanted to put it out there in case it make sense to you too.

Reply:
Thanks for your suggestion - I like the idea of giving users more flexibility. I made the update through commit 15db8d9. I think I executed this in the way you had this in mind, but let me know if you had something else in mind!

Reply:
Weird, I thought this comment had disappeared. Sorry to double up on this.